pkg/trace/writer/sender.go

```diff
@@ -210,27 +213,19 @@ outer:
 
 // Push pushes p onto the sender's queue, to be written to the destination.
 func (s *sender) Push(p *payload) {
-	for {
-		select {
-		case s.queue <- p:
-			// ok
-			s.inflight.Inc()
-			return
-		default:
-			// drop the oldest item in the queue to make room
-			select {
-			case p := <-s.queue:
-				s.releasePayload(p, eventTypeDropped, &eventData{
-					bytes: p.body.Len(),
-					count: 1,
-				})
-			default:
-				// the queue got drained; not very likely to happen, but
-				// we shouldn't risk a deadlock
-				continue
-			}
-		}
+	s.mu.RLock()
+	if s.closed {
+		s.mu.RUnlock()
+		return
 	}
+	s.mu.RUnlock()
+	select {
+	case s.queue <- p:
+	default:
+		metrics.Count("datadog.trace_agent.sender.push_blocked", 1, nil, 1)
```

Comment:
@ahmed-mez I'm going to leave this one undocumented for now. I'm not sure it's useful to customers yet. It'll mostly be useful for us until we understand what we're looking for.